### PR TITLE
Ensure that pgx generates schema on Alpine Linux

### PR DESCRIPTION
**Pull request by JamesGuthrie**
The missing *-unknown-linux-musl targets meant that pgx's magic linker
script was not used, breaking schema gen on Alpine

### CODE REVIEW

**Comment by JamesGuthrie** [comment]
We _could_ merge this change now, but if we re-run `cargo pgx schema -f <pgversion>` then it will overwrite this file, because pgx hasn't released this upstream yet.

**Comment by bitwalker** [comment]
Since you've removed the `--force-default` flag, that shouldn't be an issue right? It's only a problem if one runs that manually, unless I've misunderstood. I'll leave it up to you, but I think we're clear to merge this

**Comment by JamesGuthrie** [comment]
I've created a branch named `promscale-staging` in the https://github.com/timescale/pgx repo to track any of our changes which we're trying to upstream. It currently has the same contents as upstream `develop`. Switching this change to use that branch gives us the alpine fix which has been upstreamed but which is not in master yet. With that, this change is ready to be merged.

File: Cargo.toml
Comment (bitwalker) [inline Cargo.toml:29]: There's a slightly easier way to do this generally, using the `[patch]` section:

```toml
[patch.crates-io]
pgx = { git = "https://github.com/timescale/pgx", branch = "promscale-staging" }
pgx-macros = { git = "https://github.com/timescale/pgx", branch = "promscale-staging" }
pgx-tests = { git = "https://github.com/timescale/pgx", branch = "promscale-staging" }
```

When placed in a workspace `Cargo.toml` (or in a single crate's `Cargo.toml`), you only have to specify the override in one place, and it will replace all uses of those crates in any of the workspace crates. In this case it's 6 of one, half a dozen of the other; but probably worth following the convention anyway.

Comment (JamesGuthrie) [reply Cargo.toml:29]: TIL. Thank you!

**Review by bitwalker — APPROVED** [review]
LGTM!

**Review by jgpruitt — APPROVED** [review]

**Pull request merged**
